src/clr-angular/button/button-group/button.ts

```diff
@@ -108,6 +111,16 @@ export class ClrButton {
         }
     }
 
+    public loading: boolean;
+
+    startLoading(): void {
```

Comment:
Why wouldn't we want to move this into a reusable piece at some point? Seems unnecessary to have this boilerplate duplicated everytime someone wants to use `LoadingListener`.

Reply:
Because annotations don't have access to the injector, and dynamically modifying a component's declaration breaks AoT.

Inheritance would be the main way to do it, but inheritance is a terrible pattern for such composable features, we can't use it.

Reply:
I was thinking something along the lines of a mixin (ala Rails). But, yes, composition. If composition isn't viable in our framework of choice...

¯\_(ツ)_/¯